README.md

```diff
@@ -1 +1,162 @@
-# vue-charts
\ No newline at end of file
+# vue-charts
+
+## Installation
+* Install from npm:
+  ```
+  npm install @reside-ic/vue-charts
+  ```
+* Import into your project and register components globally or locally:
+  ```
+  import {BarChart} from "@reside-ic/vue-dynamic-form"
+  
+  // global
+  Vue.component("bar-chart", BarChart)
+  
+  // or local
+  new Vue({
+    el: '#app',
+    components: {
+      BarChart
+    }
+  })
+  
+  ```
+* Include the following css file in your app: 
+  ```
+  dist/css/style.min.css
+  ```
+  
+### Browser
+
+To use the dynamic form component directly in the browser, 
+just include `dist/js/vue-dynamic-form.min.js` on the page, after Vue and Chart.js:
+
+```
+<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.1/Chart.min.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+<script type="text/javascript" src="node_modules/@reside-ic/vue-charts/dist/vue-charts.min.js"></script>
+```
+
+In this case the component will be automatically registered. 
+
+## Components
+### BarChartWithErrors
+A simple extension of the `vue-chartjs` component which includes error bars.
+See [examples/barchart.html](examples/barchart.html).
+
+### BarChartWithFilters
+A fancier bar chart that gives the user control over what is plotted. At least 2 dimensions for filtering the data
+must be provided. 
+
+#### Props
+This component has 4 required props.
+#### chartData
+The first one is the data array itself. Array item fields must include
+* a value field
+* a field specifying what the value means
+* a lower bound for the value
+* an upper bound for the value
+* at least 2 other dimensions
+
+So this is valid data:
+
+```
+[
+    {
+        value: 0.1, indicator: "prev", e_l: 0.09, e_h: 0.11, age: "<5", sex: "1"
+    },
+    {
+        value: 0.2, indicator: "prev", e_l: 0.19, e_h: 0.21, age: "<5", sex: "2"
+    },
+    {
+        value: 0.1, indicator: "inc", e_l: 0.05, e_h: 0.15, age: "<5", sex: "1"
+    },
+    {
+        value: 0.05, indicator: "inc", e_l: 0.01, e_h: 0.07, age: "<5", sex: "2"
+    }
+]
+```
+
+and so is this:
+```
+[
+    {
+        count: 10, metric: "prev", lower_bound: 9, upper_bound: 11, year: 2001, country: "AFG"
+    },
+    {
+        count: 12, metric: "prev", lower_bound: 12, upper_bound: 13, year: 2002, country: "AFG"
+    }
+]
+```
+
+#### indicators
+The indicators prop contains metadata about the data array. It specifies the names of the 
+ required fields, and which metrics to plot.
+
+```
+[{
```

Comment:
It looks to me like we could improve this metadata a bit - `indicator_column`, `value_column`, `error_low_column` and `error_high_column` are common to the data structure so don't need to be repeated for each indicator.

I'd propose changing this to an object called `metadata` or `indicatorConfig` or something, that looks like:
```
{
        value_column: "value",
        indicator_column: "indicator",
        error_low_column: "e_l",
        error_high_column: "e_h",
        indicators: [ 
              {
                    id: "prev",
                    name: "Prevalence",
                    value: "prev"
              },
              {
                    id: "inc",
                    name: "Incidence",
                    value: "inc"
              }
       ]
}
```

Reply:
Ah right, I assumed there was some reason. Anyway I think worth opening a ticket to think about how we can optimise this interface because it's quite confusing - when I was writing the README I really struggled to explain it!

Reply:
I wonder whether this component will actually end up getting re-used, it feels very hint specific at the moment. If it was re-used I can imagine wanting to support data where there's only one indicator, or data where there are multiple indicators but no disaggregations, etc. I guess we can cross that bridge if/when we get to it.

Reply:
Difficult to predict... At least now we've broken it out of HINT so we're in a position to re-use anything in common with the next project... (even if that's not much!)